src/workerd/api/http-test.js

```diff
@@ -269,9 +269,14 @@ export const cacheMode = {
       await assertFetchCacheRejectsError('no-transform');
       await assertFetchCacheRejectsError('unsupported');
     } else {
-      await assertRequestCacheThrowsError('no-store',
-        'TypeError',
-        "Unsupported cache mode: no-store");
+      {
+        const req = new Request('https://example.org', {cache : "no-store"});
```

Comment:
Should there be a test for `no-cache` too? What about a more end-to-end test that covers the changes to `fetchImplNoOutputLock`?